be/src/vec/columns/column_array.cpp

```diff
@@ -422,7 +422,8 @@ void ColumnArray::reserve(size_t n) {
 
 //please check you real need size in data column, because it's maybe need greater size when data is string column
 void ColumnArray::resize(size_t n) {
-    get_offsets().resize(n);
+    auto last_off = get_offsets().back();
+    get_offsets().resize_fill(n, last_off);
     get_data().resize(n);
```

Comment:
Is this `get_data().resize(n)` correct? for example if this array contains 3 element s`[1, 2]` `[1, 2]` `[1, 2]`, so the inner data size is 6 instead of 3. so after doing `::resize(3)`, the inner data size became 3 and lost 3 nested element.Is this semantic right?

Reply:
I think it should not call get_data().resize(n). Just append offsets with the same value.